classes/mail/MailTemplate.inc.php

```diff
@@ -345,71 +281,6 @@ function clearRecipients($clearHeaders = true) {
 			$this->setData('headers', null);
 		}
 	}
-
-	/**
-	 * Adds a persistent attachment to the current list.
-	 * Persistent attachments MUST be previously initialized
-	 * with handleAttachments.
-	 */
-	function addPersistAttachment($temporaryFile) {
-		$this->persistAttachments[] = $temporaryFile;
-	}
-
-	/**
-	 * Handles attachments in a generalized manner in situations where
-	 * an email message must span several requests. Called from the
-	 * constructor when attachments are enabled.
-	 */
-	function _handleAttachments($userId) {
-		import('lib.pkp.classes.file.TemporaryFileManager');
-		$temporaryFileManager = new TemporaryFileManager();
-
-		$this->attachmentsEnabled = true;
-		$this->persistAttachments = array();
-
-		$deleteAttachment = Request::getUserVar('deleteAttachment');
-
-		if (Request::getUserVar('persistAttachments') != null) foreach (Request::getUserVar('persistAttachments') as $fileId) {
-			$temporaryFile = $temporaryFileManager->getFile($fileId, $userId);
-			if (!empty($temporaryFile)) {
-				if ($deleteAttachment != $temporaryFile->getId()) {
-					$this->persistAttachments[] = $temporaryFile;
-				} else {
-					// This file is being deleted.
-					$temporaryFileManager->deleteFile($temporaryFile->getId(), $userId);
-				}
-			}
-		}
-
-		if (Request::getUserVar('addAttachment') && $temporaryFileManager->uploadedFileExists('newAttachment')) {
-			$user = Request::getUser();
-
-			$this->persistAttachments[] = $temporaryFileManager->handleUpload('newAttachment', $user->getId());
-		}
-	}
-
-	function getAttachmentFiles() {
-		if ($this->attachmentsEnabled) return $this->persistAttachments;
-		return array();
-	}
-
-	/**
-	 * Delete all attachments associated with this message.
-	 * Called from send().
-	 * @param $userId int
-	 */
-	function _clearAttachments($userId) {
-		import('lib.pkp.classes.file.TemporaryFileManager');
-		$temporaryFileManager = new TemporaryFileManager();
-
-		$persistAttachments = Request::getUserVar('persistAttachments');
-		if (is_array($persistAttachments)) foreach ($persistAttachments as $fileId) {
-			$temporaryFile = $temporaryFileManager->getFile($fileId, $userId);
-			if (!empty($temporaryFile)) {
-				$temporaryFileManager->deleteFile($temporaryFile->getId(), $userId);
-			}
-		}
-	}
```

Comment:
I've removed it from the current master codebase; it was working in OJS 2.x but needs rewriting. It's really messy and not working e.g. in OMP so I removed it in the meantime.